app/scripts/metamask-controller.js

```diff
@@ -350,9 +351,7 @@ export default class MetamaskController extends EventEmitter {
         if (origin === 'metamask') {
           const selectedAddress = this.preferencesController.getSelectedAddress()
           return selectedAddress ? [selectedAddress] : []
-        } else if (
-          this.keyringController.memStore.getState().isUnlocked
-        ) {
+        } else if (this.isUnlocked()) {
```

Comment:
Is this equivalent to that which it replaced?

Reply:
Yep! `KeyringController` is in `this.memstore`, so it's included in the huge flattened state that `this.getState` returns.

Though I would suggest updating `this.isUnlocked` to return `this.keyringController.memStore.getState().isUnlocked` instead of calling `this.getState`, because the former is far more efficient. `getState` will ask for the state of _all_ controllers then flatten and combine them. Lots of work to get just one property.

Reply:
Sweet, done.